dropshot/src/error.rs

```diff
@@ -50,6 +50,123 @@ use serde::Serialize;
 use std::error::Error;
 use std::fmt;
 
+pub use crate::extractor::ExtractorError;
+pub use crate::extractor::MultipartBodyError;
+pub use crate::extractor::QueryError;
+pub use crate::extractor::StreamingBodyError;
+pub use crate::extractor::TypedBodyError;
+pub use crate::http_util::PathError;
+pub use crate::router::RouterError;
+
+#[derive(Debug, thiserror::Error)]
```

Comment:
Yup, definitely agree. I haven't written docs for most of the new stuff yet because I wanted to make sure the API direction was broadly right before spending a bunch of time documenting it, but I'm sorry if I made that hard to review!